controllers/stargate/stargate_controller.go

```diff
@@ -215,6 +215,15 @@ func (r *StargateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		}
 	}
 
+	// The following check makes sure that when Stargate is deployed on its own that the
```

Comment:
I am not sure that this is going to fix the issue. There could be just one deployment if there is only one rack in the dc; and yet, if that rack contains more than one Stargate replica, that single deployment would target more than one pod, and all these pods would get created concurrently.

Instead, I think we would need to modify `NewDeployments` so that 1) it returns a slice of deployments instead of a map, and 2) the very first deployment in the slice is guaranteed to only create one replica. Then we could wait for the first deployment to be rolled out before creating the others. Wdyt? Am I missing something?

Reply:
Basically I would see the `NewDeployments` refactored as follows:

```
// NewDeployments compute the Deployments to create for the given Stargate and CassandraDatacenter
// resources.
func NewDeployments(stargate *api.Stargate, dc *cassdcapi.CassandraDatacenter) []appsv1.Deployment {

	clusterVersion := computeClusterVersion(dc)
	seedService := computeSeedServiceUrl(dc)
	dnsPolicy := computeDNSPolicy(dc)

	racks := dc.GetRacks()
	replicasByRack := cassdcapi.SplitRacks(int(stargate.Spec.Size), len(racks))

	deployments := make([]appsv1.Deployment, 0)
	for i, rack := range racks {

		replicas := int32(replicasByRack[i])
		if replicas == 0 {
			break
		}

		template := stargate.GetRackTemplate(rack.Name).Coalesce(&stargate.Spec.StargateDatacenterTemplate)
		deploymentName := DeploymentName(dc, &rack)

		// make sure the first deployment in the slice targets exactly one replica
		if i == 0 && replicas > 1 {
			first := newDeployment(deploymentName+"-first", stargate, template, dc, rack, 1, clusterVersion, seedService, dnsPolicy)
			deployments = append(deployments, first)
			replicas--
		}

		deployment := newDeployment(deploymentName, stargate, template, dc, rack, replicas, clusterVersion, seedService, dnsPolicy)
		deployments = append(deployments, deployment)

	}
	return deployments
}
```

Reply:
@jsanda I will provide an alternate solution exploring the idea outlined above, so that we can decide tomorrow,  when you're back online, which one works best.

Reply:
@jsanda take a look at #304 when you are back.